services/download/src/main/java/app/coronawarn/server/services/download/FederationBatchProcessor.java

```diff
@@ -130,15 +133,21 @@ public void processUnprocessedFederationBatches() throws FatalFederationGatewayE
 
     while (!unprocessedBatches.isEmpty()) {
       FederationBatchInfo currentBatch = unprocessedBatches.remove();
+      cachedBatches.add(currentBatch);
       processBatchAndReturnNextBatchId(currentBatch, ERROR)
           .ifPresent(nextBatchTag -> {
-            if (config.getEfgsEnforceDateBasedDownload()) {
-              unprocessedBatches.add(new FederationBatchInfo(nextBatchTag, currentBatch.getDate()));
+            FederationBatchInfo batch = new FederationBatchInfo(nextBatchTag, currentBatch.getDate());
```

Comment:
For clarity: please indicate in the variable name that it is only a batch info, but not the batch itself

Reply:
Done. Also in the line above (nextBatchTag).